### PR TITLE
dao6 done

### PR DESCRIPTION
**Pull request by less08**

### CODE REVIEW

File: src/main/java/core/dao/jdbc/CarDaoJdbcImpl.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+                + "FROM cars c "
+                + "INNER JOIN manufacturers m ON c.manufacturer_id = m.id "
+                + "WHERE c.id = ? AND c.deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            Car car = null;
+            while (resultSet.next()) {
+                car = createCar(resultSet);
```
Comment (Max-Pochepets) [inline src/main/java/core/dao/jdbc/CarDaoJdbcImpl.java]: if I'm not mistaken, `createCar()` returns a car without assigned drivers

File: src/main/java/core/dao/jdbc/CarDaoJdbcImpl.java [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+        String query = "SELECT c.id, c.model, m.id AS manufacturer_id, m.name, m.country "
+                + "FROM cars c "
+                + "INNER JOIN manufacturers m ON c.manufacturer_id = m.id "
+                + "WHERE c.deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            List<Car> carList = new ArrayList<>();
+            while (resultSet.next()) {
+                carList.add(createCar(resultSet));
```
Comment (Max-Pochepets) [inline src/main/java/core/dao/jdbc/CarDaoJdbcImpl.java]: same

File: src/main/java/core/dao/jdbc/CarDaoJdbcImpl.java [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+        String deleteQuery = "DELETE FROM cars_drivers WHERE car_id = ?";
+        String insertQuery = "INSERT INTO cars_drivers (driver_id, car_id) VALUES (?, ?)";
+        String updateQuery = "UPDATE cars SET manufacturer_id = ?, model = ? "
+                + "WHERE id = ? AND deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement deleteStatement = connection.prepareStatement(deleteQuery);
+                 PreparedStatement insertStatement = connection.prepareStatement(insertQuery);
+                 PreparedStatement updateStatement = connection.prepareStatement(updateQuery)) {
+
+            //update car
```
Comment (Max-Pochepets) [inline src/main/java/core/dao/jdbc/CarDaoJdbcImpl.java]: for me these comments are OK, but statements names speak for themselves. you can delete comments and leave line separations if you want to highlight the stages

File: src/main/java/core/dao/jdbc/CarDaoJdbcImpl.java [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+            for (Driver driver : car.getDrivers()) {
+                insertStatement.setLong(1, driver.getId());
+                insertStatement.setLong(2, car.getId());
+                insertStatement.executeUpdate();
+            }
+
+            return car;
+
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update car ", e);
```
Comment (Max-Pochepets) [inline src/main/java/core/dao/jdbc/CarDaoJdbcImpl.java]: which car?

File: src/main/java/core/dao/jdbc/CarDaoJdbcImpl.java [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete car with id:" + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT cars.car_id, cars.model, m.manufacturer_id, "
+                + "m.manufacturer_name, m.manufacturer_country, "
+                + "d.driver_id, d.driver_name, d.licence_number FROM cars "
```
Comment (Max-Pochepets) [inline src/main/java/core/dao/jdbc/CarDaoJdbcImpl.java]:
```suggestion
                + "d.driver_id, d.driver_name, d.licence_number FROM cars c"
```
you should either use alias everywhere or don't use it at all

File: src/main/java/core/dao/jdbc/CarDaoJdbcImpl.java [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+                + "JOIN cars_drivers cd ON cars.car_id = cd.car_id "
+                + "JOIN drivers d ON d.driver_id = cd.driver_id "
+                + "JOIN manufacturers m ON cars.manufacturer_id = m.manufacturer_id "
+                + "WHERE cars.deleted = FALSE AND d.id=?";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement getFilteredStmt = connection.prepareStatement(query)) {
```
Comment (Max-Pochepets) [inline src/main/java/core/dao/jdbc/CarDaoJdbcImpl.java]:
```suggestion
                 PreparedStatement getFilteredStatement = connection.prepareStatement(query)) {
```
let's use full names, it won't get much of your time considering modern technologies

**Review by Max-Pochepets — CHANGES_REQUESTED** [review]

**Review by Max-Pochepets — APPROVED** [review]

File: src/main/java/core/dao/jdbc/CarDaoJdbcImpl.java [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+                + "WHERE c.id = ? AND c.deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            Car car = null;
+            while (resultSet.next()) {
+                car = createCar(connection, resultSet);
+            }
+            return Optional.ofNullable(car);
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/dao/jdbc/CarDaoJdbcImpl.java]:
```suggestion
            while (resultSet.next()) {
                return Optional.of(createCar(connection, resultSet));
            }
            return Optional.empty();
```

File: src/main/java/core/dao/jdbc/CarDaoJdbcImpl.java [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT c.car_id, c.model, m.manufacturer_id, "
+                + "m.manufacturer_name, m.manufacturer_country, "
+                + "d.driver_id, d.driver_name, d.licence_number FROM cars c "
+                + "JOIN cars_drivers cd ON c.car_id = cd.car_id "
+                + "JOIN drivers d ON d.driver_id = cd.driver_id "
+                + "JOIN manufacturers m ON cars.manufacturer_id = m.manufacturer_id "
+                + "WHERE c.deleted = FALSE AND d.id=?";
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/dao/jdbc/CarDaoJdbcImpl.java]: check drivers on deleted too

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**